packages/strapi-parts/src/Combobox/Combobox.js

```diff
@@ -123,6 +127,11 @@ export const Combobox = ({
   };
 
   const onInputBlur = () => {
+    if (value && !ignoreBlur.current) {
+      const valueToSet = nodes.find((node) => node.props?.value.toLowerCase() === value.toLowerCase()).props?.children;
```

Comment:
WYT of extracting this complex line (that looks the same as the line  43) on a dedicated function with a meaningful name?